ModelLibrary/Process/AttachMailToBP.cs

```diff
@@ -352,6 +367,13 @@ private void GetMails(UserInformation user, int AD_User_ID, int AD_Client_ID, in
                                         existRec = GetAttachedRecord(_tableID, Convert.ToInt32(dt.Rows[j][1]), Convert.ToInt32(uid), folderName);
                                         userOrBp = Msg.GetMsg(GetCtx(), "BusinessPartner");
                                     }
+                                    //if sender is lead
+                                    if (sender == "C_Lead")
+                                    {
+                                        _tableID = PO.Get_Table_ID("C_Lead");
+                                        existRec = GetAttachedRecord(_tableID, Convert.ToInt32(dt.Rows[j][0]), Convert.ToInt32(uid), folderName);
```

Comment:
Use Util.GetValueOfInt instead of Convert.ToInt32, as it will handle null as well